client/llb/merge.go

```diff
@@ -70,6 +70,32 @@ func (m *MergeOp) Inputs() []Output {
 	return m.inputs
 }
 
+// Merge merges multiple states into a single state. This is useful in
+// conjunction with [Diff] to create set of patches which are independent of
+// each other and can be applied in any order to a base state without affecting
```

Comment:
"any order" is confusing in here as merge does not mix the order. The order will be same as order of inputs.

Reply:
The difference between copy and merge is that copy makes a duplicate of file on top of another directory, while merge creates a directory whose contents is an overlay of 2 states on top of each other.

Reply:
Thanks, I was really getting into the weeds trying to write this one.

Reply:
I pushed an updated wording, PTAL.